src/ui/mails/Mails.js

```diff
@@ -24,6 +24,23 @@ class Mails extends React.Component {
     _fetchMails() {
         fetch('/api/mails')
             .then(data => (data.json()))
+            .then(data => {
+                data.forEach(m => {
+                    if(m.personalizations[0].dynamic_template_data) {
```

Comment:
Just looking at the first value of the `personalizations` object is kind of cumbersome, since we just drop any further entries of the object. I understand the reasons for doing that for this PR, but maybe we/I can work out a solution afterwards where we "flatten" the mails rows based on the content of the `personalizations` object.